src/ecosystem/corals/growth.jl

```diff
@@ -247,6 +272,194 @@ function bleaching_mortality!(Y::AbstractArray{Float64,2},
+        # Skip if location experiences no heat stress or there is no population
+        if dhw[loc] == 0.0 || cover[sp_sc, loc] == 0.0
+            continue
+        end
+
+        affected_pop::Float64 = cdf(dist[sp_sc, loc], dhw[loc])
+        mort_pop::Float64 = 0.0
+        if affected_pop > 0.0
+            # Calculate depth-adjusted bleaching mortality
+            mort_pop = affected_pop * depth_coeff[loc]
+
+            # Set values close to 0.0 (e.g., 1e-214) to 0.0
+            # https://github.com/JuliaLang/julia/issues/23376#issuecomment-324649815
+            if (mort_pop + one(mort_pop)) ≈ one(mort_pop)
+                mort_pop = 0.0
+            end
+        end
+
+        prop_mort[sp_sc, loc] = mort_pop
+        if mort_pop > 0.0
+            # Re-create distribution
+            d = censored(dist[sp_sc, loc]; lower=mort_pop)
+            μ::Float64 = mean(d)
+            dist_t1[sp_sc, loc] = TruncatedNormal(μ, std(d), 0.0, μ + HEAT_UB)
+
+            # Update population
+            cover[sp_sc, loc] = cover[sp_sc, loc] * (1.0 - mort_pop)
+        end
+    end
+end
+
+"""
+    _merge_distributions!(c_t, c_t1, dists, dists_t1, c_increase)
+
+Combine distributions using the weighted average approach for all size classes above 1.
+If a positive change in cover (between \$t\$ and \$t+1\$) is found for a given size
+class, the distribution of the previous size class is weighted by the difference, and the
+distribution of the current size class is weighted by [current cover - difference].
+Where a negative change has occurred, it is assumed mortalities overcame growth.
+
+# Arguments
+- `c_t` : Cover for given size class, location at timestep \$t\$
+- `c_t1` : Cover for given size class, location at timestep \$t+1\$
+- `dists` : Critical DHW threshold distribution for timestep \$t\$
+- `dists_t1` : Critical DHW threshold distribution for timestep \$t+1\$
+- `c_increase` : Cache matrix to temporarily store difference between \$c_t\$ and \$c_t1\$
+
+# Returns
+Nothing
+"""
+function _merge_distributions!(c_t, c_t1, dists, dists_t1, c_increase)::Nothing
+    # Identify size class populations that increased in cover.
+    # Assume an increase means the previous size class moved up (i.e., there was growth).
+    c_increase .= max.(c_t1 .- c_t, 0.0)
+    if all(c_increase .== 0.0)
+        return
+    end
+
+    moved = findall(c_increase[2:end] .> 0.0) .+ 1
+
+    # Calculate weights
+    w1::Vector{Float64} = c_increase ./ c_t1
+    w1[isnan.(w1)] .= 0.0
+    w2::Vector{Float64} = c_t ./ c_t1
+    w2[isnan.(w2)] .= 0.0
+
+    # Weight distributions according to their relative contributions.
+    μ1::Vector{Float64} = mean.(dists[moved.-1]) .* w1[moved.-1]
+    σ1::Vector{Float64} = std.(dists[moved.-1]) .* w1[moved.-1]
+
+    μ2::Vector{Float64} = mean.(dists[moved]) .* w2[moved]
+    σ2::Vector{Float64} = std.(dists[moved]) .* w2[moved]
+
+    dists_t1[moved] .= TruncatedNormal.(μ1 .+ μ2, σ1 .+ σ2, 0.0, (μ1 .+ μ2) .+ 20.0)
```

Comment:
What is the +20  in the truncated normal dist. here?

Reply:
This should be a constant (`HEAT_UB`). It adjusts the upper bound of the truncated distribution so we never hit the upper limit over time. Have adjusted now